src/api/app/models/maintenance_incident.rb

```diff
@@ -7,70 +7,86 @@ class MaintenanceIncident < ActiveRecord::Base
 
   def project_name
       unless self.incident_id
-        sql = ActiveRecord::Base.connection()
-        r = sql.execute( "SELECT counter FROM incident_counter WHERE maintenance_db_project_id = " + self.maintenance_db_project_id.to_s + " FOR UPDATE" ).first
-
+        r = MaintenanceIncident.exec_query(["SELECT counter FROM incident_counter WHERE maintenance_db_project_id = ? FOR UPDATE",
+                                            self.maintenance_db_project_id]).first
         if r.nil?
           # no counter exists, initialize it and select again
-          sql.execute( "INSERT INTO incident_counter(maintenance_db_project_id) VALUES('" + self.maintenance_db_project_id.to_s + "')" )
-          r = sql.execute( "SELECT counter FROM incident_counter WHERE maintenance_db_project_id = " + self.maintenance_db_project_id.to_s + " FOR UPDATE" ).first
+          MaintenanceIncident.exec_query ["INSERT INTO incident_counter(maintenance_db_project_id) VALUES('?')", self.maintenance_db_project_id]
+            
+          r = MaintenanceIncident.exec_query(["SELECT counter FROM incident_counter WHERE maintenance_db_project_id = ? FOR UPDATE",
+                                              self.maintenance_db_project_id ]).first
         end
+        
         # do an atomic increase of counter
-        sql.execute( "UPDATE incident_counter SET counter = counter+1 WHERE maintenance_db_project_id = " + self.maintenance_db_project_id.to_s )
+        MaintenanceIncident.exec_query ["UPDATE incident_counter SET counter = ? WHERE maintenance_db_project_id = ?", 
+                                        counter+1,self.maintenance_db_project_id]
         self.incident_id = r[0]
```

Comment:
where is counter coming from? and what is the problem with the counter = counter +1?